tests/testthat/test-derive_vars_merged_vaccine.R

```diff
@@ -0,0 +1,279 @@
+                    {
+                      face1 <- tibble::tribble(
+                        ~USUBJID, ~FASCAT,
+                        "ABC-1001", "SYSTEMIC",
+                        "ABC-1001", "ADMINISTRATION SITE"
+                      )
+
+                      ex1 <- tibble::tribble(
+                        ~USUBJID, ~VISIT, ~EXTRT, ~EXDOSE, ~EXSEQ,
+                        "ABC-1001","WEEK 1", "VACCINE A", 0.5, 1
+                      )
+
+
+                      expect_sys <- face1 %>% mutate(FATPTREF='VACCINATION 1') %>%
+                        left_join(ex1 %>% mutate(EXTPTREF='VACCINATION 1'), by=c('USUBJID'='USUBJID','FATPTREF'='EXTPTREF')) %>%
+                        filter(FASCAT=='SYSTEMIC')
+                        # select(-EXLAT, -EXDIR, -EXLOC)
+
+                      expect_ad <- face1 %>% mutate(FATPTREF='VACCINATION 1',
+                                                    FALAT='',
+                                                     FADIR='',
+                                                     FALOC='') %>%
+                        filter(FASCAT!='SYSTEMIC') %>%
+                        left_join(ex1 %>% mutate(EXTPTREF='VACCINATION 1',
+                                                 EXLAT='',
+                                                 EXDIR='',
+                                                 EXLOC=''), by=c('USUBJID'='USUBJID','FATPTREF'='EXTPTREF',
+                                                                 'FALAT'="EXLAT", "FADIR"="EXDIR", "FALOC"="EXLOC"))
+
+                      expected <- plyr::rbind.fill(expect_sys,expect_ad) %>% mutate(FAEVINTX='') %>%
+                        select(-VISIT) %>%
+                        mutate(FALAT='',
+                               FADIR='',
+                               FALOC='')
+
+                      actual <- derive_vars_merged_vaccine(
+                        dataset = face1,
+                        dataset_ex = ex1,
+                        dataset_supp = NULL,
+                        dataset_suppex = NULL,
+                        ex_vars = exprs(EXTRT,EXDOSE,EXSEQ)
+                      )
+
+                      expect_dfs_equal(actual, expected,keys = c("USUBJID","FASCAT"))
+                    })
+
+
+
+# testthat of derive_vars_merged_vaccine (test case 6) --------------------
+
+# testthat::test_that("derive_vars_merged_vaccine test case 6 - merge the face and ex domain on records by USUBJID and FATPTREF if the FATPTREF, FALAT, FADIR and FALOC variables are not in the face domain.
+#                     This also includes the SUPPFACE",
+#                     {
+#
+#                       face1 <- tibble::tribble(
+#                         ~STUDYID,~DOMAIN, ~USUBJID, ~FASCAT, ~FASEQ,
+#                         'ABC', 'FA', "ABC-1001", "SYSTEMIC", 1,
+#                         'ABC', 'FA', "ABC-1001", "ADMINISTRATION SITE", 2
```

Comment:
There shouldn't be any commented out code in any files.  The lintr check will fail